secrets/secrets-kubernetes/pkg/secrets/ed25519.go

```diff
@@ -0,0 +1,64 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+
+	jwt "github.com/golang-jwt/jwt/v5"
+	"github.com/nats-io/nkeys"
+)
+
+var ErrEd25519Verification = errors.New("ed25519: verification error")
+
+// SigningMethodNats implements the Ed25519 family.
+type SigningMethodNats struct{}
+
+// Specific instance for Ed25519 - nkeys edition
+var (
+	SigningMethodEd25519 *SigningMethodNats
+)
+
+func init() {
+	SigningMethodEd25519 = &SigningMethodNats{}
+	jwt.RegisterSigningMethod(SigningMethodEd25519.Alg(), func() jwt.SigningMethod {
```

Comment:
I'm glad you can do this with Go packages, because I still have yet to find a Rust crate that will let you define arbitrary key signing methods